db/database.go

```diff
@@ -545,10 +559,25 @@ func installViews(bucket base.Bucket) error {
 
 	// add all design docs from map into bucket
 	for designDocName, designDoc := range designDocMap {
-		if err := bucket.PutDDoc(designDocName, designDoc); err != nil {
-			base.Warn("Error installing Couchbase design doc: %v", err)
-			return err
+
+		//start a retry loop to put design document backing off double the delay each time
+		worker := func() (shouldRetry bool, err error, value interface{}) {
+			err = bucket.PutDDoc(designDocName, designDoc)
+			if err != nil {
+				base.Warn("Error installing Couchbase design doc: %v", err)
+			}
+			return err != nil, err, nil
 		}
+
+		sleeper := base.CreateDoublingSleeperFunc(
+			20, //MaxNumRetries
```

Comment:
I think 20 is much longer than we'll need - that ends up with the retry loop running for hours, doesn't it?  I was assuming we were looking for something on the order of 5-10 minutes, and have it report failure if it doesn't succeed by then.

An alternative would be to have the DoublingSleeperFunc have a max wait time (maybe 1 min), so that if it gets to that value, it stops doubling.